microblocks-adapter.js

```diff
@@ -65,23 +66,47 @@ class MicroBlocksDevice extends Device {
             myself.variables[index] = varName;
         };
 
-        // Unfinished. Turn single-property devices into specific things
-        /*
         if (Object.keys(deviceDescription.properties).length === 1) {
             // this is a single-property device
             var property =
                 deviceDescription.properties[
                     Object.keys(deviceDescription.properties)[0]
                 ];
+
             if (property.type === 'boolean') {
                 this.type = 'onOffSwitch';
-                // should map property "on" to actual property
-            } else if (property.type === 'number') {
+                deviceDescription.properties.on = {};
```

Comment:
I think you can do without this 1-level deep copy and instead do `deviceDescription.properties.on = deviceDescription.properties[propertyKey]` since you later delete the original property.